synapse/state/__init__.py

```diff
@@ -294,9 +294,12 @@ async def compute_event_context(
 
             # We make sure that we have a state group assigned to the state.
             if entry.state_group is None:
-                state_ids_before_event = await entry.get_state(
-                    self._state_storage_controller, StateFilter.all()
-                )
+                state_ids_before_event = None
```

Comment:
hoist this to before `if entry.state_group` to avoid doing it in both arms of the if statement?